Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -282,52 +282,113 @@ namespace OpenInfraPlatform {
 						\note	The \c IfcBSplineSurface is a subtype of \c IfcBoundedSurface.
 					*/
 
-				std::shared_ptr<carve::input::PolylineSetData> convertIfcBSplineSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBSplineSurface>& surface,
+				std::shared_ptr<carve::input::PolylineSetData> convertIfcBSplineSurface(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcBSplineSurface>& surface,
 					const carve::math::Matrix& pos) const throw(...) 
 				{
 					if (surface.expired()) {
 						throw oip::ReferenceExpiredException(surface);
 					}
 
-					// temporary call of SplineConverterT::convertIfcBSplineSurface(..)
-					SplineConverterT<IfcEntityTypesT> splineConverter(GeomSettings(), UnitConvert(), placementConverter);
-					std::shared_ptr<carve::input::PolylineSetData> polylineData;
-					splineConverter.convertIfcBSplineSurface(surface, pos, polylineData);
+					// obtain control points, attribute 3 of IfcBSplineSurface
+					const std::vector<std::vector<carve::geom::vector<3>>> controlPoints = convertIfcCartesianPoint2DVector(surface->ControlPointsList);

```

Comment:
As discussed, a better name would be `convertIfcCartesianPointVectorVector`

Reply:
Done in commit 36d839f.